content/docs/iac/using-pulumi/extending-pulumi/build-a-component.md

```diff
@@ -0,0 +1,1125 @@
+---
+title_tag: "Build a Component"
+meta_desc: "Learn the process for building a custom Pulumi Component."
+title: Build a Component
```

Comment:
Tangent - but are we planning to update tutorials like https://www.pulumi.com/tutorials/abstraction-encapsulation/

There is also a (pretty bad) https://www.pulumi.com/registry/packages/aws/how-to-guides/s3-folder-component/